Sources/NIOSSL/NIOSSLHandler.swift

```diff
@@ -469,15 +553,24 @@ public class NIOSSLHandler : ChannelInboundHandler, ChannelOutboundHandler, Remo
                     // is the result of a wonky I/O pattern and just ignore it.
                     self.plaintextReadBuffer = receiveBuffer
                     break readLoop
-                case .active:
-                    self.state = .closing(self.scheduleTimedOutShutdown(context: context))
-                case .unwrapping, .closing:
+                case .active, .outputClosed:
+                    if self.getAllowRemoteHalfClosureFromChannel(context: context) == false {
```

Comment:
Ideally we'd cache this across the call to `doFlushReadData`.

Reply:
Good point!